pkg/controller/middleware/logger.go

```diff
@@ -61,12 +57,9 @@ func PopulateLogger(originalLogger *zap.SugaredLogger) mux.MiddlewareFunc {
 			}
 
 			// On Google Cloud, extract the trace context and add it to the logger.
-			if v := r.Header.Get(googleCloudTraceHeader); v != "" && googleCloudProjectID != "" {
-				parts := strings.Split(v, "/")
-				if len(parts) > 0 && len(parts[0]) > 0 {
-					val := fmt.Sprintf("projects/%s/traces/%s", googleCloudProjectID, parts[0])
-					logger = logger.With(googleCloudTraceKey, val)
-				}
+			if id := controller.TraceIDFromContext(ctx); id != "" && googleCloudProjectID != "" {
```

Comment:
This introduces an undocumented middleware ordering issue...

Reply:
We already had one (for request ID generation). Documented both.